### ISSUE TITLE
all-in folding (after checking?) weirdness

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#522]
Gerdog:

> hand 156748 weirdness

```JPMassar was sent to Table 2380
gerdog 🍺 has come from Table 2379
ODB Phat Mack🚀🚀 has come from Table 2379
🙉pokerchimp🙈 has come from Table 2379
Dealing #156748: 6000 12000 (2000) Stud (High/Low Eight Qualifier)
Everyone else antes 1000
                           My cards are: [Ah] [9d] Ks
margaretsm antes 400 all-in
margaretsm checks
margaretsm folds due to being on vacation
1     smalltalkdan     80075 1000  [  ] [  ] Qs
2     gerdog 🍺         36400 1000  [  ] [  ] Ks
3     🎴NNR13           19225 1000  [  ] [  ] 5c
4     ODB Phat Mack🚀🚀  33300 1000  [  ] [  ] 7h
5     🙉pokerchimp🙈     42300 1000  [  ] [  ] 7s
6     JPMassar         49150 1000  [  ] [  ] Jd
7 B > tegwin           32150 1000  [  ] [  ] 3h
8    [margaretsm]      0[ 400] 
tegwin brings-in for 2000
smalltalkdan folds
It is 2000 to you
gerdog 🍺 folds
🎴NNR13 folds
ODB Phat Mack🚀🚀 folds
🙉pokerchimp🙈 folds
JPMassar calls
1    [smalltalkdan]    80075[1000] 
2    [gerdog 🍺]        36400[1000] 
3    [🎴NNR13]          19225[1000] 
4    [ODB Phat Mack🚀🚀] 33300[1000] 
5    [🙉pokerchimp🙈]    42300[1000] 
6     JPMassar         47150 3000  Jh 8h Jd
7 B   tegwin           30150 3000  Jc Kd 3h
8    [margaretsm]      0[ 400] 
1    [smalltalkdan]    80075[1000] 
2    [gerdog 🍺]        36400[1000] 
3    [🎴NNR13]          19225[1000] 
4    [ODB Phat Mack🚀🚀] 33300[1000] 
5    [🙉pokerchimp🙈]    42300[1000] 
6     JPMassar         47150 3000  Jh 8h Jd 2h
7 B   tegwin           30150 3000  Jc Kd 3h 4d
8    [margaretsm]      0[ 400] 
1    [smalltalkdan]    80075[1000] 
2    [gerdog 🍺]        36400[1000] 
3    [🎴NNR13]          19225[1000] 
4    [ODB Phat Mack🚀🚀] 33300[1000] 
5    [🙉pokerchimp🙈]    42300[1000] 
6     JPMassar         47150 3000  Jh 8h Jd 2h 3d
7 B   tegwin           30150 3000  Jc Kd 3h 4d Kc
8    [margaretsm]      0[ 400] 
tegwin: huh?
JPMassar: wtf
1    [smalltalkdan]    80075[1000] 
2    [gerdog 🍺]        36400[1000] 
3    [🎴NNR13]          19225[1000] 
4    [ODB Phat Mack🚀🚀] 33300[1000] 
5    [🙉pokerchimp🙈]    42300[1000] 
6     JPMassar         47150 3000  Jh 8h Jd 2h 3d Td
7 B   tegwin           30150 3000  Jc Kd 3h 4d Kc Js
8    [margaretsm]      0[ 400] 
1    [smalltalkdan]    80075[1000] 
2    [gerdog 🍺]        36400[1000] 
3    [🎴NNR13]          19225[1000] 
4    [ODB Phat Mack🚀🚀] 33300[1000] 
5    [🙉pokerchimp🙈]    42300[1000] 
6     JPMassar         47150 3000  Jh 8h Jd 2h 3d Td 7c
7 B   tegwin           30150 3000  Jc Kd 3h 4d Kc Js 5s
8    [margaretsm]      0[ 400] 
tegwin    Jc Kd 5s
    High  Two Pair (Kings and Jacks)   Kc Kd Jc Js 5s
     Low  none
JPMassar  Mucks
tegwin    wins 11400 for High

margaretsm was busted by tegwin in 8th place
🙉pokerchimp🙈: gg
ODB Phat Mack🚀🚀: gg'''

**Comment by ctm** [comment]
Although this was reported yesterday, I haven't looked at it yet. The description is just a copy of the hand that gerdog sent me. This looks like a bug I introduced with the new anti-zombie code. It will probably be easy to track down.

**Comment by ctm** [comment]
I'm looking at this now. I took a little time off from working on mb2 to write [convert-moves](https://github.com/ctm/moves-still-count#convert-moves) so I could continue using my Ambit 3 after Suunto stopped allowing uploads to Movescount. Electrical issues and weather have also been distracting.

**Comment by ctm** [comment]
The `margaretsm checks` was being generated because there was logic that said if someone is trying to fold when on vacation and there's a mandatory bring-in, that person should call. IOW, it assumed that if there's a mandatory bring-in and fold was being called that the action was on the player, but when I added the anti-zombie code, I introduced the possibility of folding before the cards are dealt. 

So Margaret was folded before she had cards, but the code mistakenly thought she had to call and then fold. However, both calling and folding decrements the number of players that are not all in, so when there were two players who could act it thought there was only one player who could act.

Fixed. Deploying now.
